src/libs/actions/TeachersUnite.js

```diff
@@ -0,0 +1,180 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import Navigation from '../Navigation/Navigation';
+import CONST from '../../CONST';
+import * as ReportUtils from '../ReportUtils';
+import * as OptionsListUtils from '../OptionsListUtils';
+
+let sessionEmail = '';
+let sessionAccountID = 0;
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (val) => {
+        sessionEmail = lodashGet(val, 'email', '');
+        sessionAccountID = lodashGet(val, 'accountID', 0);
+    },
+});
+
+let allPersonalDetails;
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+    callback: (val) => (allPersonalDetails = val),
+});
+
+/**
+ * @param {String} firstName
+ * @param {String} phoneOrEmail
+ * @param {String} lastName
+ */
+function referTeachersUniteVolunteer(firstName, phoneOrEmail, lastName) {
+    const reportID = CONST.TEACHERS_UNITE.PUBLIC_ROOM_ID;
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: {
+                firstName,
+                lastName,
+                phoneOrEmail,
+            },
+        },
+    ];
+    API.write(
+        'ReferTeachersUniteVolunteer',
+        {
+            reportID,
+            firstName,
+            lastName,
+            phoneOrEmail,
+        },
+        {optimisticData},
+    );
+    Navigation.dismissModal(CONST.TEACHERS_UNITE.PUBLIC_ROOM_ID);
+}
+
+/**
+ * Optimistically creates a policyExpenseChat for the school principal and passes data to AddSchoolPrincipal
+ * @param {String} firstName
+ * @param {String} email
+ * @param {String} lastName
+ */
+function addSchoolPrincipal(firstName, email, lastName) {
+    const policyName = CONST.TEACHERS_UNITE.POLICY_NAME;
+    const policyID = CONST.TEACHERS_UNITE.POLICY_ID;
+    const loggedInEmail = OptionsListUtils.addSMSDomainIfPhoneNumber(sessionEmail);
+    const reportCreationData = {};
+
+    const expenseChatData = ReportUtils.buildOptimisticChatReport([sessionAccountID], '', CONST.REPORT.CHAT_TYPE.POLICY_EXPENSE_CHAT, policyID, sessionAccountID, true, policyName);
+    const expenseChatReportID = expenseChatData.reportID;
+    const expenseReportCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(sessionEmail);
+    const expenseReportActionData = {
+        [expenseReportCreatedAction.reportActionID]: expenseReportCreatedAction,
+    };
+
+    reportCreationData[loggedInEmail] = {
+        reportID: expenseChatReportID,
+        reportActionID: expenseReportCreatedAction.reportActionID,
+    };
+
+    API.write(
+        'AddSchoolPrincipal',
+        {
+            firstName,
+            lastName,
+            email,
```

Comment:
AddSchoolPrincipal will receive `partnerUserID` for both email and phone number too

```suggestion
            partnerUserID,
```

Reply:
You mean the names of the inputs in the forms correct? If so, yeah I think it would be better to rename it to `partnerUserID` which is the correct name, but in the text labels it will be better to keep it as `email or phone number`